internal/framework/generic_resource.go

```diff
@@ -325,6 +349,15 @@ func (r *GenericResource) AwaitReady(ctx context.Context, state ResourceState, o
 		if readyErr == nil {
 			return nil
 		}
+		// Return early if resource in failed state for several iterations
```

Comment:
Do you know if this is still required with the latest CP version?
Several fixes in CP v2.6.0 improved the state transitions. In particular - [efb18cf4](https://github.com/nuodb/nuodb-control-plane/commit/efb18cf4b713c66d3018cfa3c2a6090230b6064a).

Reply:
I I did observe that the new test case fails with the latest code, but I think it is due to the fact that the initial comparison that we perform is using the result of `Read()` and not the `PUT` response. Changing it 2 iterations should be fine, but I don't think it is a big deal to leave it at 3 to be safe.

Another issue that just occurred to me is that the tests that use KWOK (K8s without real processes) take slightly longer after the timeouts change because the timeout was increased from 1s to 5s in a previous commit. I think this is not noticeable in real configurations, but it might make sense in a future change to at least use the last `PUT` response on the initial check.

Reply:
Okay. Thanks for explanation!